ra2ce/runners/direct_analysis_runner.py

```diff
@@ -48,11 +52,72 @@ def can_run(ra2ce_input: ConfigWrapper) -> bool:
         _network_config = ra2ce_input.network_config.config_data
         if not _network_config.hazard or not _network_config.hazard.hazard_map:
             logging.error(
-                "Please define a hazardmap in your network.ini file. Unable to calculate direct damages."
+                "Please define a hazard map in your network.ini file. Unable to calculate direct damages."
             )
             return False
         return True
 
+    @staticmethod
+    def _get_result_link_based(analysis: AnalysisDirectProtocol, analysis_config: AnalysisConfigWrapper,
```

Comment:
For readability you could split _get_result_link_based in smaller (inner) functions.